pytripgui/plan_vc/plan_view.py

```diff
@@ -0,0 +1,295 @@
+from PyQt5 import QtWidgets
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QTreeWidgetItem
+from pytripgui.view.gen.plan import Ui_PlanDialog
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class PlanQtView(object):
+    """
+    """
+    def __init__(self):
+        self.ui = Ui_PlanDialog()
+        self.dialog = QtWidgets.QDialog()
+
+        self.ui.setupUi(self.dialog)
+
+    def show(self):
+        self.dialog.show()
+        self.dialog.exec_()
+
+    def exit(self):
+        self.dialog.close()
+
+    def set_ok_callback(self, fun):
+        self.ui.accept_buttonBox.accepted.connect(fun)
+
+    def set_cancel_callback(self, fun):
+        self.ui.accept_buttonBox.rejected.connect(fun)
+
+    @property
+    def basename(self):    # start
+        return self.ui.basename_lineEdit.text()
+
+    @basename.getter
+    def basename(self):    # start
+        return self.ui.basename_lineEdit.text()
+
+    @basename.setter
+    def basename(self, basename):
+        self.ui.basename_lineEdit.setText(basename)
+
+    @property
+    def comment(self):
+        return self.ui.comment_lineEdit.text()
+
+    @comment.getter
+    def comment(self):
+        return self.ui.comment_lineEdit.text()
+
+    @comment.setter
+    def comment(self, comment):
+        self.ui.comment_lineEdit.setText(comment)
+
+    @property
+    def uuid(self):
+        return self.ui.uuid_lineEdit.text()
+
+    @uuid.getter
+    def uuid(self):
+        return self.ui.uuid_lineEdit.text()
+
+    @uuid.setter
+    def uuid(self, uuid):
+        self.ui.uuid_lineEdit.setText(uuid)
+
+    def add_target_roi_with_name(self, target, target_name):
+        self.ui.targetROI_comboBox.addItem(target_name, target)
+
+    def select_target_roi_to_this(self, target):
+        index_of_target = self.ui.targetROI_comboBox.findData(target, Qt.UserRole)
+        if index_of_target == -1:
+            raise Exception("Given target roi wasn't found on the list")
+        self.ui.targetROI_comboBox.setCurrentIndex(index_of_target)
+
+    def get_selected_target_roi(self):
+        return self.ui.targetROI_comboBox.currentData()
+
+    def add_oar_with_name(self, voi, voi_name):
+        oar_list = self.ui.OAR_treeWidget
```

Comment:
Does it make sense to define a variable which is used only once ?
`self.ui.OAR_treeWidget.addTopLevelItem(oar_item)` looks perfectly OK.

Reply:
done